src/solvers/lowering/byte_operators.cpp

```diff
@@ -21,6 +21,41 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/simplify_expr.h>
 #include <util/string_constant.h>
 
+/// Determine the member of maximum fixed bit width in a union type. If no
+/// member, or no member of fixed and non-zero width can be found, return
+/// nullopt.
+/// \param union_type: Type to determine the member of.
+/// \param ns: Namespace to resolve tag types.
+/// \return Pair of a componentt pointing to the maximum fixed bit-width
+///   member of \p union_type and the bit width of that member.
+static optionalt<std::pair<struct_union_typet::componentt, mp_integer>>
+find_widest_union_component(const union_typet &union_type, const namespacet &ns)
```

Comment:
Is there any particular reason this returns an optional pair, rather than just an optional `componentt` ? Given that the componentt has the the appropriate type in it and the caller could just do `pointer_offset_bits` themselves. If it's just to simplify future clients, then fine :-) But just wondering as I don't see any current clients using the second half of the pair, and having the extra `->first` or `->second` slightly reduces readability at the call sites.

Reply:
Yes, that second part is only used in #5252. I hope that's ok - I'll go ahead and merge.

Reply:
FWIW, you're of course right that the caller could compute the size themselves, but I thought duplicating work was not a good idea.

Reply:
Fine by me. I was reviewing in PR order so hadn't seen the followup PR at that point.